django/core/checks/urls.py

```diff
@@ -27,12 +27,31 @@ def check_resolver(resolver):
             warnings.extend(check_resolver(pattern))
         elif isinstance(pattern, RegexURLPattern):
             warnings.extend(check_pattern_name(pattern))
+        else:
+            # This is not a url() instance
+            warnings.extend(warn_invalid_pattern(pattern))
 
-        warnings.extend(check_pattern_startswith_slash(pattern))
+        if not warnings:
+            warnings.extend(check_pattern_startswith_slash(pattern))
 
     return warnings
 
 
+def warn_invalid_pattern(pattern):
+    """
+    Return a list containing a warning that the pattern is invalid.
+
+    Note that we cannot use describe_pattern here, because we cannot
```

Comment:
Simplify: "describe_pattern() cannot be used here because the pattern might not have..."